### DISCUSSION TITLE
Implement Dynamic Guidelines to estimate Rhein-Swimming Water Quality

### DISCUSSION THREAD

**Discussion by Butanding** [chrisrickenbacher/rhygfuehl#25] [Ideas]
I was always asking myself: When is the best time to swim in the Rhein in Basel.
So i did some research and talked to some officials. Basically, the quality of the Rhein-Water depends on two things:
1) How much did it rain the last days?
2) How much UV-Rays/Sunshine was there the last days?

It would be a cool feature, to see the sum of precipitation over the last 72h in Basel. Maybe in combination with UV-Rays of the Last 72h. Together with an explanation, what makes up for a healthy Rhein Water to swim in.

I could also help you implement it, if you give me a basic lesson on how to add a new tab or wherever you think this could be cool.

A more precise explanation can be found in the annual report on water quality: 
https://www.kantonslabor.bs.ch/dam/jcr:5053284a-cd44-490f-9cea-6c44c99aaa1d/2022-Fliesgew%C3%A4sser.pdf

The relevant part is here:

> Allgemein kann gesagt wer den, dass längere Schönwetterperioden, verbunden mit intensiver Sonneneinstrahlung, einen positiven
Effekt auf die Wasserqualität haben. Die UV-Strahlen des Sonnenlichts töten Bakterien an der Wasser oberfläche ab. Andersherum nimmt die Wasserqualität nach Gewittern und starken Regenfällen eher ab. Dann nämlich kommt es durch das Aufwühlen des Flussbettgrundes und den Eintrag von Materialien von aussen (Bäume, Blätter, Fäkalien, Abschwemmungen von landwirtschaftlich genutzten Flächen) zu einer vermehrten Belastung mit Mikroorganismen. Ausserdem stellt bei bzw. nach Regenwetterperioden auch
die Siedlungsentwässerung eine wichtige Quelle für die Belastung von Gewässern mit Mikroorganismen dar. Kläranlagen verarbeiten dann nur etwa das Doppelte des bei Trockenwetter anfallenden Wassers, so dass überschüssiges, ungereinigtes Abwasser, das mit viel Regenwasser vermischt ist, in das Gewässer gelangt.

**Comment by chrisrickenbacher** [comment]
@Butanding thanks for the idea. I was thinking a little about it and did some research:

For the measurement of the sunlight or more the intensity of the sun we could use the daily global radiation value from this datasource: https://data.bs.ch/explore/dataset/100254/table/?sort=date

And the amount of rain from this source: https://data.bs.ch/explore/dataset/100009/table/?disjunctive.name_original&disjunctive.name_custom&sort=dates_max_date

Are you then think about something like a `bathing quality` with let say three levels like bad, good and perfect? 

| bathing quality | sun | rain |
| :-------------- | :-: | :-: |
| perfect | global daily radiation over 72h above certain value | 72h no more rain then a minimal amount | 
| good | global daily radiation over 72h above certain value | 48h no more rain then a minimal amount |
| bad | else | else |

Would that make sense from your point of view? 

Do you have any idea what values we should take for:
- global radiation for perfect?
- global radiation for good?
- minimal a mount of rain for a day?

**Reply by Butanding** [reply]
Hi @chrisrickenbacher 
Thanks for your ideas. 

Why don't you just use the first source? https://data.bs.ch/explore/dataset/100254/table/?sort=date
As far as I can see, it has both, precipitation and radiation measured in Binningen.

To be honest, it can get quite complex to break down the all the factors into an easy formula.
I will forward your formula to the Department "Badewasserkontrolle" of Kanton Basel Stadt and see what they say.
I would also want to ask them about the 48/72 hours schedule.. maybe we can reduce it to 24/48 hours.
Though i think there is a slight risk, that they won't like the idea of a "simplified" version of their very complex laboratory researches of the Rhein-Water Quality. But let's ask.

Will get back to you when I get a reply.

**Comment by chrisrickenbacher** [comment]
I build a prototype with the following logic:

dailyRainTreshold = 2 mm
dailyRadiationTreshold = 170 W/m2

If maxRainLast72h < dailyRainTreshold and globalRadiationLast72h > ( dailyRadiationTreshold * 3 ) => perfect
If maxRainLast48h < dailyRainTreshold and globalRadiationLast48h > ( dailyRadiationTreshold * 2 ) => good
otherwise => bad

Now we can observe the result and tweak those thresholds a bit.

**Reply by Butanding** [reply]
Hi @chrisrickenbacher,
i like your implementation :)
During the last weeks i had some talks with people from the Kanton and they all hated the idea and do not want to give out any information or want to be connected with the project. In my eyes understandable, they are officials and the water quality barometer is just a very loose information with no guarantee.
On the other hand i find it a very suitable tool at hand and i think it has a very vital right to exist.

I would, though, add some more background information on it. Here is some ideas to think about:
1) Maybe add a tooltip explaining the idea and background how you calculate such Water Quality Barometer
2) Reference some of the Studies from the Kanton for annotation, that where the source for the idea of this implementation
3) Maybe give it a different name to reflect the "non liability" of this water quality barometer. For example: "rhygfuehl.ch Wasserqualität" or "Planschqualität" or "Wasserqualitäts-Prognose" or "Wasserqualitäts-Schätzung" or "Wasserqualitäts-Trend" or something alike.

Other than that i really like the idea.. Lets observe a little bit the quality over the next weeks and see how it evolves.

How do you like it so far?

**Comment by Butanding** [comment]
Hi @chrisrickenbacher,
long time no see on this project :) I wanted to ask why did you disable this feature actually?
I thought it was really helpful, but maybe you did it out of a reason?
Summer is starting, so would be nice to know when to swim or not to swim :)
Cheers

**Reply by chrisrickenbacher** [reply]
Hi @Butanding 

Thnks for reaching out!

I removed the old implementation because the metrics weren't accurate enough. The previous logic used static thresholds that often resulted in misleading "Perfect" or "Bad" ratings.

I've just implemented a completely new "Swimming Prognosis" logic (currently in BETA) that is much more robust. It factors in:

* Rain Memory: A 72h weighted impact score where recent rain has a higher impact on water quality than rain from two days ago.
* UV Recovery Bonus: Sunlight is now factored in as a natural disinfectant that speeds up the "clearing" of the river.
* Safety Overrides: The prognosis now includes automatic safety caps for temperature (preventing "Excellent" ratings when the water is cold enough for a cold-shock risk, or too warm, which promotes bacterial growth).

You can find the full technical breakdown of the new calculation here: https://github.com/chrisrickenbacher/rhygfuehl/blob/master/docs/prognosis-logic.md

Looking forward to your feedback!

**Comment by Butanding** [comment]
Hi Chris,
Wow, what a well-thought-out feature! I really appreciate the scientific and mathematical approach you've taken here.
I've studied your formula in detail and also worked my way through the current publications from the canton — it's quite a rabbit hole, and I find it strangely fascinating.

Interesting Publication 1: https://media.bs.ch/original_file/21798b258aba71dde3fa63134f8fc1a2a280f5c1/2025-rheinwasser.pdf

Interesting Publication 2: https://media.bs.ch/original_file/e22fc488e773274433964a019f8889be1c92c8df/2025-fliessgewaesser.pdf

Both Publications are made by Dr. Mareike Schmidt btw. Would be fun to talk to her in Person :)

Using Dr. Mareike Schmidt's studies, I tried to verify whether your formula holds up from a scientific standpoint. Here are my thoughts:

Rain Impact Score — Generally solid ✅
The research clearly supports rain as the dominant factor affecting swimming water quality. Your weighted decay approach (100% → 50% → 25%) is conceptually sound. The CSO tipping point rationale also makes sense. However, the specific weights (1.0 / 0.5 / 0.25) are currently intuitive rather than empirically calibrated. Interestingly, Dr. Schmidt's 2025 report on the Fliessgewässer includes over 100 real water sample data points with associated weather conditions. It would be worth exploring whether we could get access to that raw data and use it to train and calibrate the weights more rigorously. That could meaningfully improve the model's accuracy. But for starters i would also go with the weights you set.

UV Recovery Bonus — Needs verification ⚠️
I couldn't find any(more) direct mention of UV radiation as a factor in the studies. While cloud cover is noted as a relevant condition, the UV index specifically is not referenced. I'd recommend we revisit this parameter and either find scientific backing for it or consider removing it until it can be validated.

14°C Temperature Threshold — Good safety rule, maybe wrong category 🔄
The 14°C cold shock threshold is well-established and absolutely worth keeping. That said, I'd argue it belongs in a different context than water quality. It's really a safety indicator, not a bacterial risk indicator. My suggestion: rather than letting it influence the overall quality prognosis, we display the water temperature separately and flag it in red with a small info tag — something like "⚠️ Below 14°C — Cold shock risk, swimming not recommended" — when it drops below the threshold. This keeps the quality score focused on microbiology while still clearly communicating the safety risk.

Coliminder — Real-Time Measurement in Progress 🔬
One more thing worth highlighting: Dr. Schmidt's report mentions that the canton is currently piloting a so-called Coliminder device directly in the Rhine. This instrument is designed to measure bacterial concentrations — specifically E. coli and intestinal enterococci — automatically and in near real-time, without the usual 24–48 hour lab delay. The pilot has been running since 2025 and the results are expected to be published separately once reliable conclusions can be drawn.

Turbidity & Flow Rate — A Dead End That Validates Our Approach 🌊
It's also worth noting that Dr. Schmidt previously investigated whether water turbidity (Trübung) and flow rate (Abflussmenge) could serve as real-time proxies for microbiological water quality. The idea was appealing — both parameters are measured every 5–15 minutes and are freely available online. However, the results were clear: neither turbidity nor flow rate showed a meaningful correlation with bacterial load. The data points were too scattered, and the value ranges of the different quality classes overlapped too heavily to allow any reliable classification.
This is actually encouraging for our approach: it confirms that simple physical measurements alone aren't enough, and that a weather-based model — factoring in rainfall history and its decay over time — is a more promising direction than what the canton itself had previously attempted.

Cheers and a good start to the week ;)

**Reply by Butanding** [reply]
Another Question i asked myself: Is the Swimming Indicator Actually Relevant? What the Canton Says 📊

Absolutely. The canton's data makes a compelling case for why timely swimming warnings matter. Across 150 Rhine water samples collected between May and November 2025, the quality picture shifts dramatically depending on weather conditions. On sunny days, 90% of measurement days fell into the safe categories A or B — meaning no health risk was expected. But on overcast or rainy days, that figure flipped: more than half of those days (54%) fell into categories C or D, where a health risk cannot be ruled out or is considered possible.
In other words, on a rainy day in Basel, there's roughly a coin-flip chance that the Rhine is genuinely not safe to swim in — and currently, swimmers have no way of knowing this in real time. The canton explicitly acknowledges this gap: existing lab methods cannot provide same-day assessments, and the Coliminder project is precisely an attempt to close it.
This gives our tool a clear and legitimate purpose. We're not creating unnecessary alarm — we're filling an information gap that the canton itself has identified as a problem worth solving. Until the Coliminder delivers reliable results, our weather-proxy model is one of the only tools available that can give swimmers a meaningful, timely signal before they jump in.

**Reply by Butanding** [reply]
Hi @chrisrickenbacher,

its a long weekend, weather is bad and i had too much time to deep dive into "Oberflächengewässerberichte" of Kanton Basel Stadt. 

 Heres my quick follow-up on the UV bonus question — I went back through the 2024 and 2025 reports more carefully and also checked the wider scientific literature. Here's what I found:

**The UV bonus didn't disappear because it's wrong — it disappeared because the 2025 report has a different methodological approach**

In the 2024 Fliessgewässer report, Dr. Schmidt still explicitly mentions it:

> "Längere Schönwetterperioden, verbunden mit intensiver Sonneneinstrahlung, [haben] einen positiven Effekt auf die Wasserqualität. Die UV-Strahlen des Sonnenlichts töten Bakterien an der Wasseroberfläche ab."

In the 2025 report, this passage is completely gone. But I think I understand why: The 2025 report is not a normal annual report anymore — it's a scientific interim result from the Coliminder project with 150 Rhine samples instead of the usual ~21. The tone shifts from "here are the factors that influence quality" (popular science) to "here is what our data shows" (strictly empirical). UV radiation simply wasn't a measured variable in their study design, so it doesn't appear in the analysis.

**The 2025 data actually supports the sunshine effect indirectly**

Look at the wetterabhängige breakdown in Table 2 of the 2025 report:
- Schönwetter (20 days): 90% A/B
- Bedecktes Wetter (19 days): heavy shift toward C/D
- Regenwetter (11 days): mostly C/D

The interesting one is **bedeckt** — overcast without rain still shifts the quality distinctly worse. That's exactly the pattern you'd expect if sunshine has an independent positive effect beyond "no rain." It indirectly validates including a sunshine factor, just not as a separate "bonus" on top of a clear day.

**The science is clear that sunlight inactivates fecal indicators**

A few relevant findings from the literature:
- UV-B is the dominant cause of "natural disinfection" in surface waters; sunlight inactivation rates in rivers are >10x higher than dark inactivation rates (Sinton et al., applied & env. microbiology)
- BUT: high turbidity strongly reduces UV-B transmission. After rain, when you'd most need the UV disinfection, the water is also most turbid — so the effect is dampened exactly when it would matter most.
- A Lake Michigan beach study found that *total insolation* (global radiation) correlated better with E. coli decline than UV alone in 45 cm depth.

**Concrete suggestions for the formula**

Based on this, three thoughts:

1. **Keep the bonus, but reframe it as a Global Radiation Bonus** rather than "UV Recovery." That maps better to the data field actually available on data.bs.ch (daily global radiation from Binningen in W/m²) and is scientifically more accurate per the Lake Michigan study.

2. **Cap the bonus so it can't override the rain score.** If there was significant rain in the last 24h, the CSO overflow happened — sunshine on the surface can't undo the bacterial load *in* the water column. I'd let the bonus only operate within the "Good" tier or upgrade to "Excellent," not as a "Bad → Good" rescue.

3. **Model the 'bedeckt' case explicitly.** The 2025 report's strongest insight (to me) is that overcast without rain is still worse than sunny. That argues for a soft penalty when global radiation is low over 72h *even without rain* — not just a bonus when it's high. Cap "Excellent" at sunny days, downgrade overcast streaks to "Good" at best.

**Re: the 14°C threshold**

Still think it should be presented as a separate safety indicator rather than mixed into the microbiological quality score — for the same reason the 2025 report keeps physical and microbiological parameters cleanly separated. Different mechanism, different risk category, deserves its own visual treatment.

**One more thought — should we reach out to Dr. Schmidt?**

The situation has changed since our first attempt with the Badewasserkontrolle. The 2025 report explicitly says the canton's current methods don't allow same-day assessments, and the Coliminder pilot is their attempt to close that exact gap. Our app fills the same gap on an interim basis — it's not competing with their work, it shares the problem diagnosis. A polite inquiry asking whether the 162 data points from 2025 (with weather classification) could be shared for calibrating our weights would be the cleanest way to move from "intuitive weights" to empirically validated ones. Worth a try, I think.

Let me know what you think!

Here are some of the sources i used to write this essay:
https://media.bs.ch/original_file/80ce5c59187170a00cf7f4ffa5d59695a5a5e8f5/2024-fliessgewaesser-0.pdf

https://media.bs.ch/original_file/e22fc488e773274433964a019f8889be1c92c8df/2025-fliessgewaesser.pdf

https://www.bs.ch/news/2026-badewasserqualitaet-2025-rhein-bei-schoenwetter-gut-schwimmbaeder-weitgehend-einwandfrei

https://media.bs.ch/original_file/21798b258aba71dde3fa63134f8fc1a2a280f5c1/2025-rheinwasser.pdf

**Reply by chrisrickenbacher** [reply]
Hi @butanding

Thank you very much for your valuable input and the detailed background! I have fundamentally refactored the prognosis logic based on your feedback, check: https://github.com/chrisrickenbacher/rhygfuehl/pull/61

**The following improvements have been implemented:**

1.  Decoupled Indicators: The prognosis now separately calculates a **Water Quality Index** (microbiological risk) and a **Swimmer Safety Index** (physical risk). This provides full transparency: a "Red" status can now be clearly identified as either a hygienic risk (e.g., after rain) or a physical danger (e.g., cold shock).
2.  Radiation Bonus & Overcast Penalty: I renamed the "UV Bonus" to **"Radiation Bonus"** to correctly reflect the Global Radiation data (W/m²). Additionally, I introduced an **"Overcast Penalty"**: if the average radiation falls below 0.6 (over 72h), the quality is automatically capped at Level 1 (Discouraged) to account for the lack of natural disinfection.
3.  Refined Thresholds:
    *   Thermal Risk: If the 48h average temperature exceeds 22°C, the quality index is now capped at Level 1 due to increased microbial growth risk.
    *   Rain Tipping Point: The tolerance for today’s rain ($R_0$) in the "Good" rating has been tightened to < 1.5mm to better reflect the threshold for combined sewer overflows (CSO) in Basel.
4.  UI Transparency: These sub-indicators are now directly visible in the app when expanding the prognosis card.

**Reply by Butanding** [reply]
Hey @chrisrickenbacher i really like the new version. Great Work.
I will test evaluate the new Feature in the upcoming weeks with some real life experience and study the formula in more depth.

Also i reached out to Mrs. Dr. Schmidt and asked her, if she can send us the Data Points she collected in this Study: https://media.bs.ch/original_file/21798b258aba71dde3fa63134f8fc1a2a280f5c1/[2025-rheinwasser.pdf](https://media.bs.ch/original_file/21798b258aba71dde3fa63134f8fc1a2a280f5c1/2025-rheinwasser.pdf)

With those Data points we could test the Algorithm/Formula against tested, laboratory data that is 100% accurate and maybe fine-tune the Formula even more.
